yt/visualization/_handlers.py

```diff
@@ -0,0 +1,415 @@
+    def has_norm(self) -> bool:
+        return self._norm is not None
+
+    def _reset_norm(self):
+        if not self.has_norm:
+            return
+        mylog.warning("Dropping norm (%s)", self.norm)
+        self._norm = None
+
+    def to_float(self, val: un.unyt_quantity) -> float:
+        return float(val.to(self.display_units).d)
+
+    def to_quan(self, val) -> un.unyt_quantity:
+        if isinstance(val, un.unyt_quantity):
+            return self.ds.quan(val)
+        elif (
+            is_sequence(val)
+            and len(val) == 2
+            and isinstance(val[0], Real)
+            and isinstance(val[1], (str, un.Unit))
+        ):
+            return self.ds.quan(*val)
+        elif isinstance(val, Real):
+            return self.ds.quan(val, self.display_units)
+        else:
+            raise TypeError(f"Could not convert {val!r} to unyt_quantity")
+
+    @property
+    def display_units(self) -> un.Unit:
+        return self._display_units
+
+    @display_units.setter
+    def display_units(self, newval: Unit) -> None:
+        self._display_units = un.Unit(newval, registry=self.ds.unit_registry)
+
+    def _set_quan_attr(
+        self, attr: str, newval: Optional[Union[Quantity, float]]
+    ) -> None:
+        if newval is None:
+            setattr(self, attr, None)
+        else:
+            try:
+                quan = self.to_quan(newval)
+            except TypeError as exc:
+                raise TypeError(
+                    "Expected None, a float, or a unyt_quantity, "
+                    f"received {newval} with type {type(newval)}"
+                ) from exc
+            else:
+                setattr(self, attr, quan)
+
+    @property
+    def vmin(self) -> Optional[un.unyt_quantity]:
+        return self._vmin
+
+    @vmin.setter
+    def vmin(self, newval: Optional[Union[Quantity, float]]) -> None:
+        self._reset_norm()
+        self._set_quan_attr("_vmin", newval)
+
+    @property
+    def vmax(self) -> Optional[un.unyt_quantity]:
+        return self._vmax
+
+    @vmax.setter
+    def vmax(self, newval: Optional[Union[Quantity, float]]) -> None:
+        self._reset_norm()
+        self._set_quan_attr("_vmax", newval)
+
+    def set_dynamic_range(self, newval: Optional[float]) -> None:
+        if newval is None:
+            return
+
+        # this functionality is much more complicated to implement
+        # correctly on the library side than on the user side.
+        # Originally it is only written in support to ImagePlotContainer, to
+        # preserve backwards compatibility.
+        # The reason why it's deprecated here
+        # is to prevent this method from being reused in new code.
+
+        issue_deprecation_warning(
+            "The dynamic_range parameter is deprecated, "
+            "please pass zmin and zmax explictly.",
+            since="4.1.0",
+            # up 2 functions, one of which is doubly wrapped with decorators
+            stacklevel=6,
+        )
```

Comment:
@chrishavlin just wanted to make sure this got your attention. Hoping my justifications for deprecating this (other than frustration, that is) are satisfactory.

Reply:
actually I'm taking that back. Combining `zmin="min"` with `dynamic_range=2` for instance isn't trivial on the user side. I'm working on a better implementation.

Reply:
hmm, I don't totally see why depreciating the dynamic range option is necessary.  

You could add `dynamic_range` as another norm handler `_constraint_attrs` and delay evaluation until `get_norm`, similar to `linthresh`. e.g., I'm thinking something llke this diff for `get_norm`: https://gist.github.com/chrishavlin/839989caf7f80c2e08672e263e052eba

Reply:
Not only is it not necessary, but I changed my mind and reverted it already. I *did* reimplement it as a addition to `NormHandler._contraint_attrs`, so I think you'll like what I came up with :)
Sorry about the back and forth on this

Reply:
haha, great! no worries on the back and forth. I'll take a look tomorrow!